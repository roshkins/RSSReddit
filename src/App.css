html, body{
  width: 100%;
  height: 100%;
  background-color: #000;
  color: #AAA;
}


input {
  background-color: #000;
  border: 1px solid #AAA;
  color: #AAA;
}

a {
  color: #509;
  font-weight:600;
  font-size: 14pt;
}

div {
  outline: none;
}

/*Posts*/

.Posts{
  float: left;
  list-style-type: none;
}

.Post {
  clear: both;
  margin: 1em 0;
  overflow: hidden;
  position: relative;
}

.Error {
  background-color: #600;
}



.Score, .Thumbnail {
  float: left;
  margin: 0 1em;
}

@keyframes filter-animate {
  0% {
    filter: hue-rotate(0);
  }

  50% {
    filter: hue-rotate(360deg);
  }

  100% {
    filter: hue-rotate(0);
  }

}

.Thumbnail {
  position: relative;
  min-height: 3em;
  min-width: 150px;

}

.easter-egg {
  animation: filter-animate 5s infinite;
}


.Thumbnail::after {
  content: ' ';
  display: block;
  position: absolute;
  z-index: 2;
  background-color: #000;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
}


.Thumbnail::before {
  content: ' ';
  background: url("https://www.redditstatic.com/sprite-reddit.6Om8v6KMv28.png");
  background-position: -10px -1099px;
  background-repeat: no-repeat;
  display: block;
  position: absolute;
  z-index: 3;
  left: 0;
  width: 50px;
  height: 50px;
  margin: 10px 50px;
}

.Score {
  width: 5em;
}

.PostInfo {

  float: left;
  width: 60%;
}

/*Subreddit List*/
.SubredditList {
  float:left;
  cursor: pointer;
}
.SubredditList .Subreddit {
  float: left;
  margin-left: 2em;
  margin-top: 1pt;
}
.Subreddits {
  list-style-type: circle;
}
