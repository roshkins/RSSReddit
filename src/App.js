import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="SubredditList">
          <h1>Filter Subreddits</h1>
          <ul className="Subreddits">
            <li className="Subreddit">r/AskReddit</li>
            <li className="Subreddit">r/gadgets</li>
            <li className="Subreddit">r/sports</li>
            <li className="Subreddit">r/pics</li>
            <li className="Subreddit">r/worldnews</li>
            <li className="Subreddit">r/gaming</li>
            <li>
              <form>
                <input type="text" placeholder="r/subreddit" />
              </form>
            </li>
          </ul>

        </div>
        <ol className="Posts">
          <li className="Post">
            <div className="Score">17.1k</div>
            <img alt="thumbnail" className="Thumbnail" src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg"></img>
            <a href="#" className="title">An entire DLSR camera, disassembled.</a>
            <div>
              submitted&nbsp;
              <span className="SubmissionTime">4 hours ago</span>
               &nbsp;by&nbsp;
              <span className="Author">Idkmtbh</span>
              &nbsp;to&nbsp;
              <span className="Subreddit">r/mildlyinteresting</span>
            </div>
            <div>
              <a className="Comment" href="#">Comment</a>
            </div>
          </li>
          <li className="Post">
            <div className="Score">17.1k</div>
            <img alt="thumbnail" className="Thumbnail" src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg"></img>
            <a href="#" className="title">An entire DLSR camera, disassembled.</a>
            <div>
              submitted&nbsp;
              <span className="SubmissionTime">4 hours ago</span>
               &nbsp;by&nbsp;
              <span className="Author">Idkmtbh</span>
              &nbsp;to&nbsp;
              <span className="Subreddit">r/mildlyinteresting</span>
            </div>
            <div>
              <a className="Comment" href="#">Comment</a>
            </div>
          </li>
          <li className="Post">
            <div className="Score">17.1k</div>
            <img alt="thumbnail" className="Thumbnail" src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg"></img>
            <div className="Info">
            <a href="#" className="title">An entire DLSR camera, disassembled.</a>
              submitted&nbsp;
              <span className="SubmissionTime">4 hours ago</span>
               &nbsp;by&nbsp;
              <span className="Author">Idkmtbh</span>
              &nbsp;to&nbsp;
              <span className="Subreddit">r/mildlyinteresting</span>
              <div>
                <a className="Comment" href="#">Comment</a>
              </div>
            </div>
          </li>
        </ol>
      </div>
    );
  }
}

export default App;
