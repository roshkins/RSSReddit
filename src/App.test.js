import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('survives refactoring without changing ui', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
