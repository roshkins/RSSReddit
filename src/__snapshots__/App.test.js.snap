exports[`test survives refactoring without changing ui 1`] = `
<div
  className="App">
  <div
    className="SubredditList">
    <h1>
      Filter Subreddits
    </h1>
    <ul
      className="Subreddits">
      <li
        className="Subreddit">
        r/
        AskReddit
      </li>
      <li
        className="Subreddit">
        r/
        gadgets
      </li>
      <li
        className="Subreddit">
        r/
        sports
      </li>
      <li
        className="Subreddit">
        r/
        pics
      </li>
      <li
        className="Subreddit">
        r/
        worldnews
      </li>
      <li
        className="Subreddit">
        r/
        gaming
      </li>
      <li>
        <form>
          <input
            placeholder="r/subreddit"
            type="text" />
        </form>
      </li>
    </ul>
  </div>
  <ol
    className="Posts">
    <li
      className="Post">
      <div
        className="Score">
        17.1k
      </div>
      <img
        alt="thumbnail"
        className="Thumbnail"
        src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg" />
      <a
        className="title"
        href="#">
        An entire DLSR camera, disassembled.
      </a>
      <div>
        submitted 
        <span
          className="SubmissionTime">
          4 hours ago
        </span>
         by 
        <span
          className="Author">
          Idkmtbh
        </span>
         to 
        <span
          className="Subreddit">
          r/
          mildlyinteresting
        </span>
      </div>
      <div>
        <a
          className="Comment"
          href="#">
          Comment
        </a>
      </div>
    </li>
    <li
      className="Post">
      <div
        className="Score">
        17.1k
      </div>
      <img
        alt="thumbnail"
        className="Thumbnail"
        src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg" />
      <a
        className="title"
        href="#">
        An entire DLSR camera, disassembled.
      </a>
      <div>
        submitted 
        <span
          className="SubmissionTime">
          4 hours ago
        </span>
         by 
        <span
          className="Author">
          Idkmtbh
        </span>
         to 
        <span
          className="Subreddit">
          r/
          mildlyinteresting
        </span>
      </div>
      <div>
        <a
          className="Comment"
          href="#">
          Comment
        </a>
      </div>
    </li>
    <li
      className="Post">
      <div
        className="Score">
        17.1k
      </div>
      <img
        alt="thumbnail"
        className="Thumbnail"
        src="//b.thumbs.redditmedia.com/AwdUKfpVjz_1T7JMPfePIrS34IgEbtvEy_dWlr3bHmE.jpg" />
      <a
        className="title"
        href="#">
        An entire DLSR camera, disassembled.
      </a>
      <div>
        submitted 
        <span
          className="SubmissionTime">
          4 hours ago
        </span>
         by 
        <span
          className="Author">
          Idkmtbh
        </span>
         to 
        <span
          className="Subreddit">
          r/
          mildlyinteresting
        </span>
      </div>
      <div>
        <a
          className="Comment"
          href="#">
          Comment
        </a>
      </div>
    </li>
  </ol>
</div>
`;
